File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>


int get_height(void);
void print_hashes(int n);

int main(void)
{
    int height;
    // prompt user to enter the height
    height = get_height();

    
    for (int row = 1; row < height + 1; row++)
    {
        // print gap before hashes
        for (int col = 0; col < height - row; col++)
        {
            printf(" ");
        }
        // print left hashes
        print_hashes(row);

        // print gap between hashes
        printf("  ");

        // print right hashes
        print_hashes(row);


        printf("\n");
    }
}

// prompt user to enter height until between 1 and 8
int get_height(void)
{
    int height;

    do
    {
        height = get_int("Height (between 1 and 8): ");
    }
    while (height < 1 || height > 8);

    return height;
 

}

// print n hashes
void print_hashes(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}

File: pset6/mario/more/mario.py
import cs50


def main():
    height = 0
    # Prompt user to enter the height
    while height < 1 or height > 8:
        height = cs50.get_int("Height ")

    for row in range(height):

        # Print gap before hashes
        for col in range(height - row - 1):
            print(" ", end="")

        # Print left hashes
        print_hashes(row)

        # Print gap between hashes
        print("  ", end="")

        # Print right hashes
        print_hashes(row)

        print()


def print_hashes(n):
    for i in range(n + 1):
        print("#", end="")


main()